src/Extension/CoreExtension.php

```diff
@@ -1300,6 +1302,65 @@ function twig_test_iterable($value)
     return $value instanceof \Traversable || \is_array($value);
 }
 
+/**
+ * Checks if a variable is a sequence.
+ *
+ *    {# evaluates to true if the foo variable is a list or a traversable object #}
+ *    {% if foo is sequence %}
+ *        {# ... #}
+ *    {% endif %}
+ *
+ * @param mixed $value A variable
+ *
+ * @return bool true if the value is a sequence
+ */
+function twig_test_sequence($value)
+{
+    return $value instanceof \Traversable || (\is_array($value) && \array_is_list($value));
+}
+
+
+/**
+ * Checks if a variable is a mapping.
+ *
+ *    {# evaluates to true if the foo variable is an array but not a list #}
+ *    {% if foo is mapping %}
+ *        {# ... #}
+ *    {% endif %}
+ *
+ * @param mixed $value A variable
+ *
+ * @return bool true if the value is a mapping
+ */
+function twig_test_mapping($value)
+{
+    return \is_array($value) && ((0 === \count($value)) || !\array_is_list($value));
```

Comment:
Any `stdClass` is a mapping. That's the type returned by `json_decode` by default for objects. It's also the only way to have an empty mapping.

This test would be more useful if it was different than a `<var> is not mapping`.

Given this [definition of this test in Jinja](https://jinja.palletsprojects.com/en/2.11.x/templates/#mapping), the implementation should check that none of the key are numeric.
> mapping(value)
> Return true if the object is a mapping (dict etc.).

Reply:
numeric keys that are not sequentials would still be a mapping IMO.